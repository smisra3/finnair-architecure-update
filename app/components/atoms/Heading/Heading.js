// @flow
import type { Node } from 'react';
import styles from './Heading.style';
import styled from 'styled-components';

type Props = {
  children: Node,
  className: string,
  type?: string,
  inheritedStyles?: string,
};

const Heading = ({
  children, className, type, inheritedStyles, ...others
}: Props): Node => (
  <h1 className={className} {...others}>
    {children}
  </h1>
);

Heading.defaultProps = {
  inheritedStyles: '',
  type: 'h3',
};

export default styled(Heading)`${styles}`;
