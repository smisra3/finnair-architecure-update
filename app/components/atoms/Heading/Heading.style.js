import { css } from 'styled-components';

export default css`
  letter-spacing: 1px;
  font-weight: normal;
  font-family: ${props => props.theme.primaryFontFamily};
  font-size: 20px;
`;
