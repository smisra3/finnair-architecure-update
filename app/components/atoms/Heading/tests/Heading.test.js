import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Heading from '..';

const props = {
  children: 'Sample',
  className: '',
};
describe('Heading Component', () => {
  let heading;

  test('should render correctly and match snapshot', () => {
    heading = shallow(<Heading {...props} />);
    expect(heading).toMatchSnapshot();
  });

  test('should render with styles correctly and match snapshot', () => {
    heading = renderer.create(<Heading {...props} />).toJSON();
    expect(heading).toMatchSnapshot();
  });
});
