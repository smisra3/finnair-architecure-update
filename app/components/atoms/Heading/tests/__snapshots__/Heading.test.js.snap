// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component should render correctly and match snapshot 1`] = `
.c0 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-weight: normal;
  font-size: 20px;
}

<Heading
  className="c0"
  inheritedStyles=""
  type="h3"
>
  Sample
</Heading>
`;

exports[`Heading Component should render with styles correctly and match snapshot 1`] = `
.c0 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-weight: normal;
  font-size: 20px;
}

<h1
  className="c0"
>
  Sample
</h1>
`;
